modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -1121,6 +1121,349 @@ definition {
 		AppBuilderAdmin.validateIfSearchItemIsNotFound(itemName = "Apple");
 	}
 
+	@description = "Validate empty list state of Apps listing screen"
+	@priority = "5"
+	test ImplementRulesForFieldSets {
+		property portal.acceptance = "true";
+
+		AppBuilderAdmin.addCustomObject(
+			customObjectName = "my custom object",
+			isCreatingFormView = "true",
+			isCreatingViaPlusIcon = "true");
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Text");
+
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Select from List");
+
+		for (var number : list "1,2,3") {
+			FormViewBuilder.editOptionsRow(
+				optionName = "Select${number}",
+				optionValue = "Select ${number}",
+				rowNumber = "${number}");
+		}
+		FormViewBuilder.switchToAdvancedTab();
+		FormViewBuilder.selectSelectOptionPredefined(selectOption = "Select 3");
+
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Single Selection");
+
+		for (var number : list "1,2,3") {
+			FormViewBuilder.editOptionsRow(
+				optionName = "Task${number}",
+				optionValue = "Task ${number}",
+				rowNumber = "${number}");
+		}
+		FormViewBuilder.switchToAdvancedTab();
+		FormViewBuilder.selectSelectOptionPredefined(selectOption = "Task 3");
+
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Multiple Selection");
+
+		for (var number : list "1,2,3") {
+			FormViewBuilder.editOptionsRow(
+				optionName = "Multiple${number}",
+				optionValue = "Multiple ${number}",
+				rowNumber = "${number}");
+		}
+		FormViewBuilder.switchToAdvancedTab();
+		FormViewBuilder.selectSelectWithMultipleOptionsPredefined(
+			fieldName = "Multiple Selection",
+			selectOptionsList = "Multiple 3");
+
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Numeric");
+
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Upload");
+
+		Click(locator1 = "FormViewBuilder#RULES_TAB");
+
+		Click(locator1 = "FormViewBuilder#CREATE_NEW_RULE_BUTTON");
+
+		Type(
```

Comment:
Hi @calvinctvo , please use FormViewBuilder.setRuleName macro.

Reply:
please, update all the occurrences of this action replacing by the setRuleName macro.